publish-service/src/main/java/com/ericsson/eiffel/remrem/publish/controller/ProducerController.java

```diff
@@ -193,65 +192,98 @@ public ResponseEntity generateAndPublish(@ApiParam(value = "message protocol", r
-            ResponseEntity<String> response = restTemplate.postForEntity(generateUrl,
-                    entity, String.class, generateURLTemplate.getMap(msgProtocol, msgType));
-
-            if (response.getStatusCode() == HttpStatus.OK) {
-                log.info("The result from REMReM Generate is: " + response.getStatusCodeValue());
-
-                // publishing requires an array if you want status code
-                String responseBody = "[" + response.getBody() + "]";
-                MsgService msgService = PublishUtils.getMessageService(msgProtocol, msgServices);
-
-                log.debug("mp: " + msgProtocol);
-                log.debug("body: " + responseBody);
-                log.debug("user domain suffix: " + userDomain + " tag: " + tag + " routing key: " + routingKey);
-                if (msgService != null && msgProtocol != null) {
-                    rmqHelper.rabbitMqPropertiesInit(msgProtocol);
-                }
-                synchronized(this) {
-                    SendResult result = messageService.send(responseBody, msgService, userDomain, tag, routingKey);
-                    return new ResponseEntity(result, messageService.getHttpStatus());
-                }
+        List<Map<String, Object>> responseEvents = new ArrayList<>();
+        for (JsonObject eventJson : events) {
+            Map<String, Object> eventResponse = new HashMap<>();
+            String bodyJsonOut = null;
+            if (parseData) {
+                // -- parse params in incoming request -> body -------------
+                EventTemplateHandler eventTemplateHandler = new EventTemplateHandler();
+                JsonNode parsedTemplate = eventTemplateHandler.eventTemplateParser(eventJson.toString(), msgType);
+                bodyJsonOut = parsedTemplate.toString();
+                log.info("Parsed template: " + bodyJsonOut);
             } else {
-                return response;
+                bodyJsonOut = eventJson.toString();
             }
-        }
-        catch (RemRemPublishException e) {
+
+            HttpHeaders headers = new HttpHeaders();
+            headers.setContentType(MediaType.APPLICATION_JSON_UTF8);
+            HttpEntity<String> entity = new HttpEntity<>(bodyJsonOut, headers);
+            EnumSet<HttpStatus> getStatus = EnumSet.of(HttpStatus.SERVICE_UNAVAILABLE, HttpStatus.UNAUTHORIZED, HttpStatus.NOT_ACCEPTABLE, HttpStatus.EXPECTATION_FAILED, HttpStatus.INTERNAL_SERVER_ERROR, HttpStatus.UNPROCESSABLE_ENTITY);
+
+            try {
+                String generateUrl = generateURLTemplate.getUrl() + "&failIfMultipleFound=" + failIfMultipleFound
+                        + "&failIfNoneFound=" + failIfNoneFound + "&lookupInExternalERs=" + lookupInExternalERs
+                        + "&lookupLimit=" + lookupLimit + "&okToLeaveOutInvalidOptionalFields=" + okToLeaveOutInvalidOptionalFields;
+                ResponseEntity<String> response = restTemplate.postForEntity(generateUrl,
```

Comment:
Why send just one vent to generate service? There's generate endpoint accepting array of events. This restores the former behaviour: a network connection per event.